src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,105 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import React, { useState, useEffect } from 'react';
+
+interface PeopleFiltersProps {
+  persons: Person[];
+  onFilterChange: (filter: 'all' | 'male' | 'female') => void;
+
+  onClick: (century: string | 'all' | null) => void;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
The `onChange` prop name might be misleading here since it suggests a generic change event handler, but it actually represents a search input change handler. Consider renaming it to something more descriptive like `onSearchChange`.